### ISSUE TITLE
Properly marshal event time to RFC-3339 format

### ISSUE THREAD

**Issue by Cali0707** [cloudevents/sdk-go#987]
In event_marshal.go, `time.String()` is currently used to marshal the time to a string: https://github.com/cloudevents/sdk-go/blob/ba92a5c9a1e2508d855930049999ccd75636f0cb/v2/event/event_marshal.go#L76-L79

However, in the go `time.Time` documentation, it says about the method `time.Time.String()`:
> The returned string is meant for debugging; for a stable serialized representation, use t.MarshalText, t.MarshalBinary, or t.Format with an explicit format string. 

In the Spec, it is required that the time is serialized to RFC-3339, so we should be using `time.Time.MarshalText` instead

**Comment by duglin** [comment]
Or time.Format() with RFC3339 - wanna submit a PR?

**Comment by Cali0707** [comment]
Happy to open a PR, but I know @MaryamTaj has been wanting to make some contributions so I'll see if she is interested in this one first :)

**Comment by MaryamTaj** [comment]
@duglin I would love to work on this!

**Comment by embano1** [comment]
Commented in the PR to discuss whether this must be considered a breaking change.

**Comment by duglin** [comment]
@Cali0707 is this actually a problem on the wire?
I agree that time.String() wouldn't be correct, but the code uses: eventContext.Time.String() - which has a custom String() func. So are you sure this is actually an issue? Are you seeing serialized CloudEvents with weird timestamps?

**Comment by duglin** [comment]
In fact, when I look at things like this: https://github.com/cloudevents/sdk-go/blob/74ac76d56f0d899827f940b1a19079014f90e749/v2/event/event_marshal_test.go#L82

<img width="419" alt="Image" src="https://github.com/user-attachments/assets/fe72a5a4-347e-4d6a-90ca-1440bfc0542d" />

it sure looks like it's serializing it, and in this case expecting it, in the right format

**Comment by embano1** [comment]
Yes, seems the SDK already does this correctly (based on the comments):

https://github.com/cloudevents/sdk-go/blob/74ac76d56f0d899827f940b1a19079014f90e749/v2/types/timestamp.go#L15

**Comment by duglin** [comment]
Going to close this issue and corresponding PR
